roles/ceph-common/templates/ceph.conf.j2

```diff
@@ -32,45 +32,53 @@ mon initial members = {% for host in groups[mon_group_name] %}
+  mon host = {% for host in groups[mon_group_name] %}{{ hostvars[host]['ansible_all_{{ ip_version }}_addresses'] | ipaddr(monitor_address_block) | first }}{% if not loop.last %},{% endif %}{% endfor %}
 {% elif groups[mon_group_name] is defined %}
-mon host = {% for host in groups[mon_group_name] %}
+mon host = {% for host in groups[mon_group_name] -%}
              {% set address = hostvars[host]['monitor_address'] if hostvars[host]['monitor_address'] is defined else monitor_address %}
              {% set interface = hostvars[host]['monitor_interface'] if hostvars[host]['monitor_interface'] is defined else monitor_interface %}
-             {% if interface != "interface" %}
-               {% for key in hostvars[host].keys() %}
+             {% if interface != "interface" -%}
+               {% for key in hostvars[host].keys() -%}
                  {% if hostvars[host][key]['macaddress'] is defined and hostvars[host][key]['device'] is defined and hostvars[host][key]['device'] == interface -%}
+                   {% if ip_version == 'ipv4' -%}
                     {{ hostvars[host][key][ip_version]['address'] }}
+                   {%- elif ip_version == 'ipv6' -%}
+                    [{{ hostvars[host][key][ip_version] | map(attribute='address') | list | ipaddr(public_network) | first }}]
+                   {%- endif %}
                  {%- endif %}
-               {% endfor %}
+               {%- endfor %}
              {% elif address != "0.0.0.0" -%}
                {{ address }}
              {%- endif %}
-             {%- if not loop.last %},{% endif %}
-           {% endfor %}
+             {% if not loop.last -%},{%- endif %}
+           {%- endfor %}
 {% endif %}
 {% endif %}
 {% if containerized_deployment %}
 fsid = {{ fsid }}
 {% if groups[mon_group_name] is defined %}
-mon host = {% for host in groups[mon_group_name] %}
+mon host = {% for host in groups[mon_group_name] -%}
              {% set interface = ["ansible_",ceph_mon_docker_interface]|join %}
              {% if containerized_deployment -%}
-                {{ hostvars[host][interface]['ipv4']['address'] }}
+               {% if ip_version == 'ipv4' -%}
+                 {{ hostvars[host][interface][ip_version]['address'] }}
+               {%- elif ip_version == 'ipv6' -%}
+                 [{{ hostvars[host][interface][ip_version] | map(attribute='address') | list | ipaddr(public_network) | first }}]
+               {%- endif %}
              {%- elif hostvars[host]['monitor_address'] is defined -%}
                 {{ hostvars[host]['monitor_address'] }}
              {%- elif monitor_address != "0.0.0.0" -%}
                 {{ monitor_address }}
              {%- endif %}
-             {%- if not loop.last %},{% endif %}
-           {% endfor %}
+             {% if not loop.last -%},{%- endif %}
+           {%- endfor %}
 {% endif %}
 {% endif %}
 
-{% if public_network is defined %}
+{% if public_network is defined and ip_version == 'ipv4' %}
```

Comment:
As far as I know, you can't pass this parameter in the ceph.conf when deploying with ipv6